docs/guides/app_model.md

```diff
@@ -0,0 +1,273 @@
+```
+
+### Commands should *not* be confused with the public napari API
+
+While it is conceivable that plugins might need/want to refer to one of these
+napari commands by its string id, it is **not** currently a goal that napari
+end-users would execute any of these commands by their ID.  There should always
+be a "pure python" way to import a napari object and call it.  Commands mostly
+serve as a way to reference some functionality that needs to be exposed in the
+GUI.
+
+## Menus
+
+All napari menus will also have a string id (e.g. `'napari/file'`, or
+`'napari/layers/context'`), which should be declared as a member of the
+`napari._app_model.constants.MenuId` enum.  Internally, an instance of this enum
+should be used instead of the string literal when referring to a menu.
+
+Menus are not limited to the visible menus in the application mene bar, they can
+appear anywhere, such as the layerlist context menu that appears when a user
+right-clicks on the layer list. They may also be used to create a toolbar (a set
+of clickable buttons that execute a command). For example, the mode buttons in
+each of the layer controls could be represented as menus with a set of commands.
+
+One of the benefits of the abstraction provided by `app-model` is that actual Qt
+Menu objects become simple to construct:
+
+```python
+>>> from app_model.backends.qt import QModelMenu
+>>> from napari._app_model.constants import MenuId
+
+# create a QMenu with all of the commands registered in the
+# layerlist context menu
+>>> menu = QModelMenu(menu_id=MenuId.LAYERLIST_CONTEXT, app='napari')
+
+>>> print([i.text() for i in menu.actions()])
+[
+    'Toggle visibility',
+    '',
+    'Convert to Labels',
+    'Convert to Image',
+    'Convert data type',
+    '',
+    'Duplicate Layer',
+    'Split Stack',
+    'Split RGB',
+    'Merge to Stack',
+    'Projections',
+    '',
+    'Link Layers',
+    'Unlink Layers',
+    'Select Linked Layers'
+]
+```
+
+For details on how menus are grouped into sections, and how commands are ordered
+within each section, see the `group` and `order` attributes in the
+[`app_model.types.MenuRule`
+documentation](https://app-model.readthedocs.io/en/latest/types/#app_model.types.MenuRule)
+
+## Keybindings
+
+`app-model` has an [extensive internal representation](https://app-model.readthedocs.io/en/latest/keybindings/) of Key codes, and
+combinations of key press events (including *chorded* key press sequences such
+as `Cmd+K` *followed by* `Cmd+M`).
+
+We don't yet use them internally, but they will provide independence from
+vispy's key codes, and have a nice `IntEnum` api that allows for declaration of
+keybindings in namespaced way that avoids usage of strings:
+
+```python
+>>> from app_model.types import KeyCode, KeyMod
+
+>>> ctrl_m = KeyMod.CtrlCmd | KeyCode.KeyM
+
+>>> ctrl_m
+<KeyCombo.CtrlCmd|KeyM: 2078>
+```
+
+## Actions
+
+The "complete" representation of a command, along with it's optional placement
```

Comment:
```suggestion
The "complete" representation of a command, along with its optional placement
```

Reply:
my most common grammatical error  😔  I even know what it's supposed to be, but I do it wrong more often than not! :joy: